boot.py

```diff
@@ -190,27 +195,39 @@ def on_post_window_command(self, window: sublime.Window, command_name: str, args
                 sublime.set_timeout(panel_manager.update_log_panel)
 
 
-class LspOpenLocationCommand(sublime_plugin.TextCommand):
+class LspOpenLocationCommand(LspWindowCommand):
     """
     A command to be used by third-party ST packages that need to open an URI with some abstract scheme.
     """
 
     def run(
         self,
-        _: sublime.Edit,
         location: Union[Location, LocationLink],
         session_name: Optional[str] = None,
         flags: int = 0,
-        group: int = -1
+        group: int = -1,
+        event: Optional[dict] = None
     ) -> None:
+        if event:
+            modifier_keys = event.get('modifier_keys')
+            if modifier_keys:
+                if 'primary' in modifier_keys:
+                    flags |= sublime.ADD_TO_SELECTION | sublime.SEMI_TRANSIENT | sublime.CLEAR_TO_RIGHT
+                elif 'shift' in modifier_keys:
+                    flags |= sublime.ADD_TO_SELECTION | sublime.SEMI_TRANSIENT
         sublime.set_timeout_async(lambda: self._run_async(location, session_name, flags, group))
 
+    def want_event(self) -> bool:
+        return True
+
     def _run_async(
-        self, location: Union[Location, LocationLink], session_name: Optional[str], flags: int = 0, group: int = -1
+        self, location: Union[Location, LocationLink], session_name: Optional[str], flags: int, group: int
     ) -> None:
-        manager = windows.lookup(self.view.window())
-        if manager:
-            manager.open_location_async(location, session_name, self.view, flags, group) \
+        if session_name:
+            self.session_name = session_name
```

Comment:
Doesn't seem right to set this as a class property. I believe that would mean that the command becomes specific to that session from that point and won't be available for other sessions anymore because `is_enabled` will only get sessions that match `self.session_name`.

Can't this just use `self.session_by_name` instead?

Reply:
Oh. Good catch, that would be an ugly bug. Should be fixed in https://github.com/sublimelsp/LSP/pull/2151/commits/30a93550ee1637cd3b7197518aeebe8f07adf460.